integrations/wallets/metamask.md

```diff
@@ -45,7 +45,160 @@ Moonbase Alpha TestNet:
+
+MetaMask injects a global Ethereum API into websites users visit at `window.ethereum`, which allows the websites to read and request the users' blockchain data. You'll be using the Ethereum provider to walk your users through the process of adding Moonbase Alpha as a custom network. In general, you will have to:
+
+- Check if the Ethereum provider exists and if it's MetaMask
+- Request the user's account address
+- Add Moonbase Alpha as a new chain
+
+This guide is divided into two sections. First, it'll cover adding a button that will be used to trigger MetaMask to pop-up and connect to Moonbase Alpha. The second part of the guide will create the logic for connecting the user to MetaMask. This way when you click the button you can actually test the functionality as you go through the guide.
+
+### Checking Prerequisites
+
+To add the Connect MetaMask button you'll need a JavaScript project and the MetaMask browser extension installed for local testing.
+
+It's recommended to use MetaMask's `detect-provider` utility package to detect the provider injected at `window.ethereum`. The package handles detecting the provider for the MetaMask extension and MetaMask Mobile. To install the package in your JavaScript project, run:
+
+```
+npm install @metamask/detect-provider
+```
+### Add a Button
+
+You'll start off by adding a button that will be used to connect MetaMask to Moonbase Alpha. You want to start with the button so when you create the logic in the next step you can test the code as you make your way through the guide. 
+
+The function we will create in the next section of the guide will be called `configureMoonbaseAlpha`. So the button on click should call `configureMoonbaseAlpha`.
+
+```html
+<button onClick={configureMoonbaseAlpha()}>Connect to Moonbase Alpha</button>
+```
+
+### Add Logic
+
+Now that you have created the button, you need to add the `configureMoonbaseAlpha` function that will be used on click. 
+
+1. Detect the provider at `window.ethereum` and check if it's MetaMask. If you want a simple solution you can directly access `window.ethereum`. Or you can use MetaMask's `detect-provider` package and it will detect the provider for MetaMask extension and MetaMask Mobile for you.
+```javascript
```

Comment:
Add space for readability (of the .md file, not the article itself)

Reply:
if I add a space the 2 will turn into a 1, and the 3 will turn into a 2, and the formatting will get messed up 😕

Reply:
Ok I understand now what you mean hehehe - we can leave it as it is then